examples/basics/gloo/spatial_filters.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Example demonstrating spatial filtering using spatial-filters fragment shader.
+Left and Right Arrow Keys toggle through available filters.
+"""
+
+import os
+import numpy as np
+
+from vispy.io import _data_dir
+from vispy import gloo
+from vispy import app
+
+# create 5x5 matrix with border pixels 0, center pixels 1
+# and other pixels 0.5
+I = np.zeros(25).reshape((5, 5)).astype(np.float32)
+I[1:4, 1::2] = 0.5
+I[1::2, 2] = 0.5
+I[2, 2] = 1.0
+
+# loading interpolation kernel
+kernel = np.load(os.path.join(_data_dir, 'spatial-filters.npy'))
```

Comment:
I agree this is not so nice -- users shoudn't need private functions/attributes (names preceded by `_`) to do things. Perhaps we should add a `vispy.io.get_spatial_filters()` function.